koku/masu/database/trino_sql/gcp/openshift/populate_daily_summary/1_resource_matching_by_cluster.sql

```diff
@@ -39,19 +39,28 @@ INSERT INTO hive.{{schema | sqlsafe}}.managed_gcp_openshift_daily_temp (
     month,
     day
 )
+WITH cte_years AS (
+    SELECT
+    DISTINCT year
+    FROM gcp_line_items_daily
+    WHERE usage_start_time >= {{start_date}}
+    AND usage_start_time <= {{end_date}}
+    AND source = {{cloud_provider_uuid}}
+),
 WITH cte_months AS (
```

Comment:
Good catch on thinking about the year. That December to Jan transition would of sucked. However, you forgot to remove the initial `WITH` here. 

However, if I can make a suggestion, you could do this as a single query if you change your filters to a join:
```
WITH cte_usage_date_partitions as (
    select 
        year, 
        month 
    from gcp_line_items_daily 
    where usage_start_time >= DATE('2025-06-20') 
    and usage_start_time <= DATE('2025-07-03') 
    group by year, month
)
SELECT gcp.month, gcp.year FROM gcp_line_items_daily as gcp
JOIN cte_usage_date_partitions ym ON gcp.year = ym.year AND gcp.month = ym.month
GROUP BY gcp.month, gcp.year;
```
Result:
```
 month | year
-------+------
 06    | 2025
 07    | 2025
(2 rows)
```

Reply:
Haha! Good catch on the with! Copy paste fail for sure, I was wondering about a single query lets do it!! Thanks Cody.